lib/srv/regular/sshserver_test.go

```diff
@@ -1411,13 +1411,53 @@ func TestClientDisconnect(t *testing.T) {
 	require.NoError(t, clt.Close())
 }
 
+func testHelperGetWaitForServerStart(t *testing.T, s *Server, config *ssh.ClientConfig) func() bool {
+	return func() bool {
+		clt, _ := ssh.Dial("tcp", s.Addr(), config)
+		if clt != nil {
+			require.NoError(t, clt.Close())
+			require.ErrorIs(t, clt.Wait(), net.ErrClosed)
+			return true
+		}
+		return false
+	}
+}
+
+func testHelperGetWaitForNumberOfConns(t *testing.T, limiter *limiter.Limiter, token string, num int64) func() bool {
+	return func() bool {
+		connNumber, err := limiter.GetNumConnection(token)
+		require.NoError(t, err)
+		return connNumber == num
+	}
+}
+
+type fakeClock struct {
+	clock clockwork.FakeClock
+}
+
```

Comment:
We should add a `var _ timetools.TimeProvider = (*fakeClock)(nil)` here so it's more clear why we're wrapping `clockwork.FakeClock`.

Reply:
`clockwork.FakeClock` is an interface, so I'd expect it to behave like a reference type in almost all cases.